fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/JS/A32NX_Simvars.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * taken from the AAU1 G3000
+ * The purpose is to reduce the amount of string allocations caused by `.toLowerCase()`
+ */
+
+const latlonaltRegEx = new RegExp(/latlonalt/i);
+const latlonaltpbhRegex = new RegExp(/latlonaltpbh/i);
+const pbhRegex = new RegExp(/pbh/i);
+const pid_structRegex = new RegExp(/pid_struct/i);
+const xyzRegex = new RegExp(/xyz/i);
+const stringRegex = new RegExp(/string/i);
+const boolRegex = new RegExp(/boolean|bool/i);
+const numberRegex = new RegExp(/number/i);
+const defaultSource = '';
+
+SimVar.GetSimVarValue = (name, unit, dataSource = defaultSource) => {
+    try {
+        if (simvar) {
+            let output;
+            const registeredID = SimVar.GetRegisteredId(name, unit, dataSource);
+
+            if (registeredID >= 0) {
+                if (numberRegex.test(unit)) {
+                    output = simvar.getValueReg(registeredID);
+                } else if (stringRegex.test(unit)) {
+                    output = simvar.getValueReg_String(registeredID);
+                } else if (latlonaltRegEx.test(unit)) {
```

Comment:
**issue**:A pattern like `/latlonalt/i` matches any string which contains this sequence of characters. Thus the `/latlonalt/i` will also match `"latlonaltpbh"`, which is probably not what you want to happen.
**suggestion**: Start with the most extensive regex first.

Reply:
The anchors ^ and $ can also be handy.

Reply:
I have decide to use the anchors now to match the exact string, as other values won't make sense anyway. Please check if that's better now :)